### PR TITLE
Issue #34 Enhance Docker Security and Best Practices

### PR DESCRIPTION
**Pull request by myselfdeepdas**
This pull request introduces improvements to enhance the security and follow best practices when running Docker containers for the OBA server and the transit data bundler.

### Changes Made:

1. **oba/Dockerfile:**
    - Create a non-root user ('oba') for improved security.
    - Switch to the non-root user for subsequent Dockerfile steps.
    - Refactor the Dockerfile to align with Docker best practices.

2. **bundler/Dockerfile:**
    - Create a non-root user ('oba') for enhanced security.
    - Switch to the non-root user for the remaining Dockerfile steps.
    - Adjust the Dockerfile according to best practices.

### Context:

The changes aim to minimize the potential security risks associated with running containers as the root user. Following best practices, we've created a dedicated non-root user for the containers, reducing the attack surface and enhancing the overall security posture.

Please review and merge these changes to ensure a more secure and reliable deployment of the OBA server and the transit data bundler containers.

Thank you!

### CODE REVIEW

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
These Dockerfiles will no longer work after the changes that are in this PR. It almost looks like the output of ChatGPT with the comments saying "Continue with the rest of the Dockerfile as the non-root user" in the Dockerfiles

**Comment by myselfdeepdas** [comment]
Check it out—I made some changes

File: oba/Dockerfile [hunk 1]
```diff
@@ -16,8 +19,6 @@ WORKDIR /oba/libs
 
 # OBA WAR and JAR files
 RUN wget "https://repo.camsys-apps.com/releases/org/onebusaway/onebusaway-transit-data-federation-webapp/${OBA_VERSION}/onebusaway-transit-data-federation-webapp-${OBA_VERSION}.war"
-RUN wget "https://repo.camsys-apps.com/releases/org/onebusaway/onebusaway-api-webapp/${OBA_VERSION}/onebusaway-api-webapp-${OBA_VERSION}.war"
```
Comment (Neo2308) [inline oba/Dockerfile:19]: @myselfdeepdas Removing these wget statements seems to be unintentional. Could you revert?

**Comment by aaronbrethorst** [comment]
Superseded by #55

**Pull request closed without merging**